### PR TITLE
Remove f000 from atmos rocoto tasks for replay cases

### LINKED ISSUE #2771: Remove atmos_prod f000 rocoto task when OFFSET_START_HOUR is greater than 0
### What is wrong?

When OFFSET_START_HOUR is greater than 0, the atmos_prod f000 task is still added as a rocoto task. This is an issue because if the OFFSET_START_HOUR is greater than 0, then the atmos_prod f000 task will never be triggered, which results in a stalled atmos_prod task. This may cause downstream tasks to never be triggered, particularly tasks that depend on all lead times to complete for the atmosphere component. 

### What should have happened?

The atmos_prod f000 task should be removed when OFFSET_START_HOUR is greater than 0. 

### What machines are impacted?

All or N/A

### Steps to reproduce

1. Set `REPLAY_ICS` to `YES`.
2. Set `OFFSET_START_HOUR` to a value greater than `0`. 
3. Make sure `atmos` is included as a component. 

### Additional information

Below is a screenshot of an example of the atmos_prod f000 task not being triggered. In this example, `OFFSET_START_HOUR` is 3. 

Screenshot from 2024-07-17 15-22-16

### Do you have a proposed solution?

One possible solution is to have an `FHMIN` specifically for the atmosphere, such as `FHMIN_ATMOS`. It should be noted that the wave component has its own `FHMIN`, which is called `FHMIN_WAV`. When `OFFSET_START_HOUR` is greater than 0, `FHMIN_ATMOS` should be set to `OFFSET_START_HOUR`, while the FHMIN for the ocean and ice should be 0.

### PR DESCRIPTION
**Pull request by EricSinsky-NOAA**
# Description

The main purpose of this PR is to remove the f000 from atmosphere-related rocoto tasks when `REPLAY_ICS` is set to `YES`. In cases where `REPLAY_ICS` is `YES` and `OFFSET_START_HOUR `is greater than `0`, it becomes necessary to have the first lead hour set to `OFFSET_START_HOUR ` for the atmosphere-related rocoto tasks. For example, when `OFFSET_START_HOUR ` is set to `3`, then the minimum lead time for the atmos_prod and atmos_ensstat rocoto tasks needs to be 3 and the minimum lead time for the ocean_prod rocoto task needs to be 6 (assuming `FHOUT_OCN` is 6). This PR makes this rocoto workflow set up possible by removing 0 from fhrs for atmosphere-related tasks in `gefs_tasks.py` when replaying. 

This PR also moves where f000 is being removed for the ocean_prod and ice_prod tasks. The if-block that performs this f000 removal has been moved from `tasks.py` to `gefs_tasks.py` and `gfs_tasks.py`.

This PR is a prerequisite for PR #2755. 

Refs #2771 
Resolves #2771 

# Type of change

- Bug fix (fixes something broken)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

This PR has been tested on WCOSS2 using a C384 GEFS case with `REPLAY_ICS` set to `YES` and `OFFSET_START_HOUR `set to `3`. 

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by EricSinsky-NOAA** [comment]
A C48_S2SWA_gefs CI test case has successfully completed. Also successfully ran a special case where `FHMIN_GFS=3`,  `FHMIN=0` and `REPLAY_ICS=YES`. Marking as ready for review.

**Comment by aerorahul** [comment]
@EricSinsky-NOAA I sent an email discussing the options.  Let's chat offline.

**Review by aerorahul — DISMISSED** [review]
lgtm

**Comment by EricSinsky-NOAA** [comment]
Thank you, @aerorahul. I ran the C48_S2SWA_gefs CI test case, which has successfully completed.

File: workflow/rocoto/gefs_tasks.py [hunk 1]
```diff
@@ -270,6 +270,17 @@ def _atmosoceaniceprod(self, component: str):
                      'maxtries': '&MAXTRIES;'}
 
         fhrs = self._get_forecast_hours('gefs', self._configs[config], component)
+
+        # when replaying, atmos component does not have fhr 0, therefore remove 0 from fhrs
+        local_config = self._configs[config].deepcopy()
+        is_replay = local_config['REPLAY_ICS']
```
Comment (WalterKolczynski-NOAA) [inline workflow/rocoto/gefs_tasks.py]: I don't think we need to make a deep copy here. We're not changing the configs, just reading them. We can use the original for that.
```suggestion
        is_replay = self._configs[config]['REPLAY_ICS']
```

File: workflow/rocoto/gefs_tasks.py [hunk 2]
```diff
@@ -314,6 +325,13 @@ def atmos_ensstat(self):
                      'maxtries': '&MAXTRIES;'}
 
         fhrs = self._get_forecast_hours('gefs', self._configs['atmos_ensstat'])
+
+        # when replaying, atmos component does not have fhr 0, therefore remove 0 from fhrs
+        local_config = self._configs['atmos_ensstat'].deepcopy()
+        is_replay = local_config['REPLAY_ICS']
```
Comment (WalterKolczynski-NOAA) [inline workflow/rocoto/gefs_tasks.py]: Same here
```suggestion
        is_replay = self._configs['atmos_ensstat']['REPLAY_ICS']
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: workflow/rocoto/gefs_tasks.py (hunk 1 above)
Comment (EricSinsky-NOAA) [reply workflow/rocoto/gefs_tasks.py]: Sure. Deep copy has been removed here and for atmos_ensstat.

**Review by aerorahul — APPROVED** [review]

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved

**Pull request merged**